noir/noir-repo/aztec_macros/src/lib.rs

```diff
@@ -1186,6 +1205,24 @@ fn create_mark_as_initialized() -> Statement {
     )))
 }
 
+/// Creates a check for internal functions ensuring that the caller is self.
+///
+/// ```noir
+/// assert(context.msg_sender() == context.this_address(), "Function can only be called internally");
+/// ```
+fn create_internal_check(fname: &str) -> Statement {
```

Comment:
Is this always required? i thought the kernel did this?

Reply:
Plan is to remove it from the kernel, so we simplify the structure of each function.